app/routers/meds.py

```diff
@@ -0,0 +1,364 @@
+from datetime import date, datetime, time, timedelta
+from typing import Dict, List, Tuple
+
+from fastapi import APIRouter
+from fastapi.param_functions import Depends
+from sqlalchemy.orm.session import Session
+from starlette.requests import Request
+from starlette.responses import RedirectResponse, Response
+
+from app.database.models import Event, User
+from app.dependencies import get_db, templates
+from app.internal.utils import (create_model, get_current_user,
+                                get_time_from_string)
+from app.main import app
+
+
+router = APIRouter(
+    prefix='/meds',
+    tags=['meds'],
+    dependencies=[Depends(get_db)],
+)
+
+MAX_EVENT_QUANTITY = 50
+
+FORM_TUPLE = Tuple[str, time, int, time, time, time, time, str, datetime,
+                   datetime]
+
+ERRORS = {
+    'finish': 'Finish Date must must be later than or equal to Start Date',
+    'max': 'Maximal Interval must must be larger than or equal to Minimal \
+        Interval',
+    'amount': 'Interval between Earliest Reminder and Latest Reminder not \
+        long enough for Daily Amount with Minimal Interval',
+    'quantity': 'Total number of reminders can\'t be larger than '
+                + f'{MAX_EVENT_QUANTITY}. Please lower the daily amount, or '
+                + 'choose a shorter time period.'
+}
+
+
+def trans_form(form: Dict[str, str]) -> FORM_TUPLE:
+    """Converts all form data to useable types and return as a Tuple.
+
+    Args:
+        form (dict(str, str)): Form to translate.
+
+    Returns:
+        tuple(...): All converted form data.
+                        name (str) - Medication name.
+                        first (time | None) - First dose time, if given.
+                        amount (int) - Daily dosage.
+                        early (time) - Earliest reminder time.
+                        late (time) - Latest reminder time.
+                        minimum (time) - Minimal interval between reminders.
+                        maximum (time) - Maximal interval between reminders.
+                        note (str) - Additional description.
+                        start (datetime) - First reminder date and time.
+                        end (datetime) - Last reminder date and time.
+    """
+    name = form['name']
+    start_date = get_time_from_string(form['start'])
+    first = get_time_from_string(form['first'])
+    end_date = get_time_from_string(form['end'])
+    amount = int(form['amount'])
+    early = get_time_from_string(form['early'])
+    late = get_time_from_string(form['late'])
+    minimum = get_time_from_string(form['min'])
+    maximum = get_time_from_string(form['max'])
+    first_time = first if first else early
+    start = datetime.combine(start_date, first_time)
+    if late <= early:
+        end_date += timedelta(1)
```

Comment:
shouldnt it be timedelta(days=1) ?

Reply:
This would be a more "polite" and clear way to write it, but since `days` is the first parameter of `datetime.timedelta` it is not required to explicitly specify it.

Reply:
You are both right. I personally would love to see the `days` argument here, it'll make it much easier to understand.